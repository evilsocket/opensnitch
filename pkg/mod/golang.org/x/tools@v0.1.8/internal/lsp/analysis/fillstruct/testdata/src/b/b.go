package fillstruct

type B struct {
	ExportedInt   int
	unexportedInt int
}
