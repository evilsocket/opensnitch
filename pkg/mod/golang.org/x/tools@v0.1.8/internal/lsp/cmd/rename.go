// Copyright 2019 The Go Authors. All rights reserved.
// Use of this source code is governed by a BSD-style
// license that can be found in the LICENSE file.

package cmd

import (
	"context"
	"flag"
	"fmt"
	"io/ioutil"
	"os"
	"path/filepath"
	"sort"

	"golang.org/x/tools/internal/lsp/diff"
	"golang.org/x/tools/internal/lsp/protocol"
	"golang.org/x/tools/internal/lsp/source"
	"golang.org/x/tools/internal/span"
	"golang.org/x/tools/internal/tool"
	errors "golang.org/x/xerrors"
)

// rename implements the rename verb for gopls.
type rename struct {
	Diff     bool `flag:"d" help:"display diffs instead of rewriting files"`
	Write    bool `flag:"w" help:"write result to (source) file instead of stdout"`
	Preserve bool `flag:"preserve" help:"preserve original files"`

	app *Application
}

func (r *rename) Name() string      { return "rename" }
func (r *rename) Usage() string     { return "<position> <new name>" }
func (r *rename) ShortHelp() string { return "rename selected identifier" }
func (r *rename) DetailedHelp(f *flag.FlagSet) {
	fmt.Fprint(f.Output(), `
Example:

  $ # 1-based location (:line:column or :#position) of the thing to change
  $ gopls rename helper/helper.go:8:6 Foo
  $ gopls rename helper/helper.go:#53 Foo

	gopls rename flags are:
`)
	f.PrintDefaults()
}

// Run renames the specified identifier and either;
// - if -w is specified, updates the file(s) in place;
// - if -d is specified, prints out unified diffs of the changes; or
// - otherwise, prints the new versions to stdout.
func (r *rename) Run(ctx context.Context, args ...string) error {
	if len(args) != 2 {
		return tool.CommandLineErrorf("definition expects 2 arguments (position, new name)")
	}
	conn, err := r.app.connect(ctx)
	if err != nil {
		return err
	}
	defer conn.terminate(ctx)

	from := span.Parse(args[0])
	file := conn.AddFile(ctx, from.URI())
	if file.err != nil {
		return file.err
	}
	loc, err := file.mapper.Location(from)
	if err != nil {
		return err
	}
	p := protocol.RenameParams{
		TextDocument: protocol.TextDocumentIdentifier{URI: loc.URI},
		Position:     loc.Range.Start,
		NewName:      args[1],
	}
	edit, err := conn.Rename(ctx, &p)
	if err != nil {
		return err
	}
	var orderedURIs []string
	edits := map[span.URI][]protocol.TextEdit{}
	for _, c := range edit.DocumentChanges {
		uri := fileURI(c.TextDocument.URI)
		edits[uri] = append(edits[uri], c.Edits...)
		orderedURIs = append(orderedURIs, string(uri))
	}
	sort.Strings(orderedURIs)
	changeCount := len(orderedURIs)

	for _, u := range orderedURIs {
		uri := span.URIFromURI(u)
		cmdFile := conn.AddFile(ctx, uri)
		filename := cmdFile.uri.Filename()

		// convert LSP-style edits to []diff.TextEdit cuz Spans are handy
		renameEdits, err := source.FromProtocolEdits(cmdFile.mapper, edits[uri])
		if err != nil {
			return errors.Errorf("%v: %v", edits, err)
		}
		newContent := diff.ApplyEdits(string(cmdFile.mapper.Content), renameEdits)

		switch {
		case r.Write:
			fmt.Fprintln(os.Stderr, filename)
			if r.Preserve {
				if err := os.Rename(filename, filename+".orig"); err != nil {
					return errors.Errorf("%v: %v", edits, err)
				}
			}
			ioutil.WriteFile(filename, []byte(newContent), 0644)
		case r.Diff:
			diffs := diff.ToUnified(filename+".orig", filename, string(cmdFile.mapper.Content), renameEdits)
			fmt.Print(diffs)
		default:
			if len(orderedURIs) > 1 {
				fmt.Printf("%s:\n", filepath.Base(filename))
			}
			fmt.Print(string(newContent))
			if changeCount > 1 { // if this wasn't last change, print newline
				fmt.Println()
			}
			changeCount -= 1
		}
	}
	return nil
}
