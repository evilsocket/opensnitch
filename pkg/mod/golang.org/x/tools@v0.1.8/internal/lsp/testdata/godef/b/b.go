package b

import (
	myFoo "golang.org/x/tools/internal/lsp/foo" //@mark(myFoo, "myFoo"),godef("myFoo", myFoo)
	"golang.org/x/tools/internal/lsp/godef/a"   //@mark(AImport, re"\".*\"")
)

type Embed struct {
	*a.A
	a.I
	a.S
}

func _() {
	e := Embed{}
	e.Hi()      //@hoverdef("Hi", AHi)
	e.B()       //@hoverdef("B", AB)
	e.Field     //@hoverdef("Field", AField)
	e.Field2    //@hoverdef("Field2", AField2)
	e.Hello()   //@hoverdef("Hello", AHello)
	e.Hey()     //@hoverdef("Hey", AHey)
	e.Goodbye() //@hoverdef("Goodbye", AGoodbye)
}

type aAlias = a.A //@mark(aAlias, "aAlias")

type S1 struct { //@S1
	F1     int //@mark(S1F1, "F1")
	S2         //@godef("S2", S2),mark(S1S2, "S2")
	a.A        //@godef("A", AString)
	aAlias     //@godef("a", aAlias)
}

type S2 struct { //@S2
	F1   string //@mark(S2F1, "F1")
	F2   int    //@mark(S2F2, "F2")
	*a.A        //@godef("A", AString),godef("a",AImport)
}

type S3 struct {
	F1 struct {
		a.A //@godef("A", AString)
	}
}

func Bar() {
	a.AStuff()  //@godef("AStuff", AStuff)
	var x S1    //@godef("S1", S1)
	_ = x.S2    //@godef("S2", S1S2)
	_ = x.F1    //@godef("F1", S1F1)
	_ = x.F2    //@godef("F2", S2F2)
	_ = x.S2.F1 //@godef("F1", S2F1)

	var _ *myFoo.StructFoo //@godef("myFoo", myFoo)
}

const X = 0 //@mark(bX, "X"),godef("X", bX)
