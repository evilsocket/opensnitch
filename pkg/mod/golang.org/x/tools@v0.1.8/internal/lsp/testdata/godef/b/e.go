package b

import (
	"fmt"

	"golang.org/x/tools/internal/lsp/godef/a"
)

func useThings() {
	t := a.Thing{}      //@mark(bStructType, "ing")
	fmt.Print(t.Member) //@mark(bMember, "ember")
	fmt.Print(a.Other)  //@mark(bVar, "ther")
	a.Things()          //@mark(bFunc, "ings")
}

/*@
godef(bStructType, Thing)
godef(bMember, Member)
godef(bVar, Other)
godef(bFunc, Things)
*/

func _() {
	var x interface{}      //@mark(eInterface, "interface{}")
	switch x := x.(type) { //@hoverdef("x", eInterface)
	case string: //@mark(eString, "string")
		fmt.Println(x) //@hoverdef("x", eString)
	case int: //@mark(eInt, "int")
		fmt.Println(x) //@hoverdef("x", eInt)
	}
}
