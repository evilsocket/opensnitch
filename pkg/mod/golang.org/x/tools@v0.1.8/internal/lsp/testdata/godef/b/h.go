package b

import . "golang.org/x/tools/internal/lsp/godef/a"

func _() {
	// variable of type a.A
	var _ A //@mark(AVariable, "_"),hoverdef("_", AVariable)

	AStuff() //@hoverdef("AStuff", AStuff)
}
