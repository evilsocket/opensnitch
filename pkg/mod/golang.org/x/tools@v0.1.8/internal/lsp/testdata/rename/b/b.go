package b

var c int //@rename("int", "uint")

func _() {
	a := 1 //@rename("a", "error")
	a = 2
	_ = a
}

var (
	// Hello there.
	// Foo does the thing.
	Foo int //@rename("Foo", "Bob")
)

/*
Hello description
*/
func Hello() {} //@rename("Hello", "Goodbye")
