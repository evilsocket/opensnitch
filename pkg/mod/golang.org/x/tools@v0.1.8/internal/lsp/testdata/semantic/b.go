package semantictokens //@ semantic("")

func f(x ...interface{}) {
}

func weirâ°€d() { /*ðŸ˜€*/ // comment
	const (
		snil   = nil
		nil    = true
		true   = false
		false  = snil
		cmd    = `foof`
		double = iota
		iota   = copy
		four   = (len(cmd)/2 < 5)
		five   = four
	)
	f(cmd, nil, double, iota)
}

/*

multiline */ /*
multiline
*/
type AA int
type BB struct {
	AA
}
type CC struct {
	AA int
}
type D func(aa AA) (BB error)
type E func(AA) BB
