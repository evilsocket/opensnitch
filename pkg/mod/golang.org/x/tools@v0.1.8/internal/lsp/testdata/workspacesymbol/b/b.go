package b

var RandomGopherVariableB = "b" //@symbol("RandomGopherVariableB", "RandomGopherVariableB", "Variable", "", "b.RandomGopherVariableB")

type RandomGopherStructB struct { //@symbol("RandomGopherStructB", "RandomGopherStructB", "Struct", "", "b.RandomGopherStructB")
	Bar int //@mark(bBar, "Bar"), symbol("Bar", "Bar", "Field", "RandomGopherStructB", "b.RandomGopherStructB.Bar")
}
