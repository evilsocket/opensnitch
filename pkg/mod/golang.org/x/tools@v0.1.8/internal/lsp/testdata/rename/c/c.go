package c

import "golang.org/x/tools/internal/lsp/rename/b"

func _() {
	b.Hello() //@rename("Hello", "Goodbye")
}
