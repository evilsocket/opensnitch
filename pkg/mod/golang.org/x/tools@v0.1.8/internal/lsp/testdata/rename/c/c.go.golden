-- Goodbye-rename --
b.go:
package b

var c int //@rename("int", "uint")

func _() {
	a := 1 //@rename("a", "error")
	a = 2
	_ = a
}

var (
	// Hello there.
	// Foo does the thing.
	Foo int //@rename("Foo", "Bob")
)

/*
Goodbye description
*/
func Goodbye() {} //@rename("Hello", "Goodbye")

c.go:
package c

import "golang.org/x/tools/internal/lsp/rename/b"

func _() {
	b.Goodbye() //@rename("Hello", "Goodbye")
}

